src/FastBoard.cpp

```diff
@@ -1000,6 +1036,14 @@ int FastBoard::string_size(int vertex) {
     return m_stones[m_parent[vertex]];
 }
 
+/*
+   Actual (real) liberties of the string the specified vertex position belongs to.
+   Implemented using union-find.
```

Comment:
That's rather misleading. The calculation is done incrementally, union-find is only used for locating the parent vertex. And even so, I don't think it's the pure union find that requires recursing on find-parent.

Reply:
I was going off of what @fishcu said. Reduced it to just a single line until a better description can be found.